webpack/assets/javascripts/react_app/components/SearchBar/index.js

```diff
@@ -1,21 +1,95 @@
-import { connect } from 'react-redux';
-import { bindActionCreators } from 'redux';
-import SearchBar from './SearchBar';
-import { selectAutocompleteSearchQuery } from '../AutoComplete/AutoCompleteSelectors';
-import { setAutocompleteSearchQuery } from '../AutoComplete/AutoCompleteActions';
+import React, { useState, useEffect } from 'react';
+import { isEmpty } from 'lodash';
+import PropTypes from 'prop-types';
+import { SearchAutocomplete } from './SearchAutocomplete';
+import { useAPI } from '../../common/hooks/API/APIHooks';
+import Bookmarks from '../PF4/Bookmarks';
+import { changeQuery } from '../../common/urlHelpers';
+import { STATUS } from '../../constants';
+import { noop } from '../../common/helpers';
 
-const mapStateToProps = (
-  state,
-  {
-    data: {
-      autocomplete: { id },
-    },
-  }
-) => ({
-  searchQuery: selectAutocompleteSearchQuery(state, id),
-});
+const SearchBar = ({
+  data: {
+    autocomplete: { url, searchQuery } = { url: '' },
+    controller,
+    bookmarks,
+    disabled,
+  },
+  initialQuery,
+  onSearch,
+  onSearchChange,
+}) => {
+  const [search, setSearch] = useState(initialQuery || searchQuery || '');
+  const { response, status, setAPIOptions } = useAPI('get', url, {
+    params: { search },
+  });
+  useEffect(() => {
+    if (searchQuery !== search) {
+      setSearch(searchQuery || '');
+      setAPIOptions({ params: { search: searchQuery || '' } });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
same comment as above

Reply:
Used this way https://beta.reactjs.org/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes